engine/src/main/java/com/arcadedb/serializer/BinaryComparator.java

```diff
@@ -396,6 +396,9 @@ public static boolean equalsString(final String buffer1, final String buffer2) {
     if (buffer1 == null || buffer2 == null)
       return false;
 
+    if (buffer1.equals("") && buffer2.equals(""))
```

Comment:
I could be more efficient doing `if( buffer1.isEmpty() && buffer2.isEmpty() )`

Reply:
Done.